### DISCUSSION TITLE
thetetriseffect

### DISCUSSION THREAD

**Discussion by giscus** [jamesstephenbrown/NZSWebsite#16] [Announcements]
# thetetriseffect

Exploration of the Tetris Effect and the development of 3D Tetris

https://nonzerosum.games/thetetriseffect.html

**Comment by FreeFallAnnie** [comment]
I got to 14100! And was working out the visuals before I fell asleep 😅 Thanks for that :P but truly thanks, it's really cool to see complexity in this way... You nail it!

Sparks my thinking... what applications might this have? what strategies might be best? and why?... .continued sparking.

**Reply by jamesstephenbrown** [reply]
Nice! And you achieved The Tetris Effect! Thanks for your comments
It's interesting to think of strategies, I found myself falling into 2D Tetris mode and neglecting the other depth levels, and then if you think about it from the perspective of a game like Block Out (another 3D Tetris game) the strategy is to fill layer by layer - but I think this is largely because Block Out suffers from the obscuring effect... please share any strategies you devise :)

**Reply by jamesstephenbrown** [reply]
Also, the mobile version will be released this week - subscribe for the update, otherwise I'll be putting it up on the socials.

**Reply by FreeFallAnnie** [reply]
A strategy I just used was to start using the closest view, then when I was able to see a deeper layer forming, I could save my thinking by converting to think about the next layer depth instead. This gave me more time to think about the closer layer too. If I panicked, I could use a failsafe of pushing back the problem piece so they were easier to deal with.... (Thought crosses my mind, low entropy and high entropy are simpler... hmmm >.>)

I guess this might be slicing up the new pattern into multiple 2D grids? Which is similar to how we look at spacetime - processing by piecing together the frames.... (if I have that connection correct).

Today, I ended up with a new high score 89800.... Guess my subconscious did some work!

(Completely up to you but I accidently hit the caps lock and end up unable to move the pieces, is there any way you could add the capitals - A,S,X,D,W -to the controls? All good if not :D)

**Reply by jamesstephenbrown** [reply]
Hey Annie :) Nice score! Thanks for the pointer about CAPS, I've fixed that now.

And yes, I get what you're saying about slicing 3D space up into 2D layers, when you think about it we do that in the real world, because we are looking at 3D space from one perspective, so we only get a 2D picture - well 2 2D pictures at slightly different perspectives that help us form a 3D model in our minds. I was hoping that would be the effect of in the game too, that we'd magically gain a 3D model in our mind.

Though that's got me thinking, it would be entirely possible to do a stereoscopic version of the game where we'd actually perceive the depth in a way that was more natural.

**Reply by FreeFallAnnie** [reply]
I like your thinking! Very curious what a stereoscopic version would look like. 🕵️‍♀️

Would be cool to have a training mode where we could pause the game and are able to toggle between versions, to compare. Or a guide for each of the shapes. (Though it is fun to explore them!)
It's an interesting thought experiment either way.

I admit, I am using this game to improve my multidimensional visualisation skills for my own layered project. Massively appreciate your efforts! 😅

**Reply by jamesstephenbrown** [reply]
Oh wow really? What is you project?

And actually a pause button is not such a bad idea, I mean, obviously people could use it to "cheat" but what would that even really mean? After all there's no leaderboard.

**Reply by FreeFallAnnie** [reply]
People could use it to cheat, and it could take away the angle of competition is (I feel) part of the appeal. A fix could be to make it so the training mode doesn't have points. That could help keep the challenge for the player.

Ah. My project. 🫣 Well.

I have a habit of liking and creating frameworks that are recursive (particularly in the area of cognition and communication) -the project I am playing with includes creating modules using the concept of neural nets. It is really tricky to visualise them clearly and quickly though I am determined to do so (😵‍💫<--- actually me) . 

Last year I turned to considering mind mapping with scalar, vector, matrix, and tensor descriptions to assist with visualising mental blocks with the goal of being able to build them up so I can break them down and move them about. Between last year and now I did discover psychology using quantum cognition as standard practise. So considering how everything connects.... I am in a perpetual game of mental Tetris... where the blocks are rubix cubes 🫠

What stumped me with the scalar, vector etc. visuals is that I couldn't fill in all the gaps of the shapes and wanted to be able to move around rather than building thoughts up 'linearly' to chunk them. Which was an issue I found with memory palaces. So that visualisation method took a back burner. (Instead I went to creating multidimensional characters! ... They are fun. And another story.. multiple stories 😅.)

Hadn't revisited the initial idea of visualising the multiple layers until this game. 

It's really helpful, because the shapes aren't needing the gaps to be filled, and can maintain scalar (0D), vector (1D) matrix (2D) and tensor (3D) frames of reference that then connect to objects of similar structure - what I love is that it helps with nonlinearity, and can allow for missing information. My goal here is to allow the mind to wander, accounting for recursion, backfill, and dynamic behaviour.

For example (warning this is a terrible example hoping you get my thinking) - take a basic framework that includes a why and/or a how and/or a what.
Each of these can be separated and make for 0D thinking.

I.e. 'What' am I doing.... -> eating a cookie. = scalar

But say I want to envision more information so I decide...

What is my What = eating a cookie
What is my How = using Nan's recipe
What is my Why = to seek Comfort

This becomes a vector.
O
O
O

On Layer 1.
Layer 1 is the What Layer.
This vector would have the largest ring.
O

Go to Layer 2.
Layer 2 is the How Layer.
Medium ring.
o

Layer 3.
Layer 3 is the Why Layer.
Smallest ring.
.

It might make sense for Layer 2 (the how layer) to include
How am I doing my What = 'eating' a cookie
How am I doing my How = using Nan's recipe
How am I doing my Why = to seek Comfort
o
o
o

But on layer 3 something interesting happens. 

Why am I doing my What = eating a cookie
Why am I doing my How = using Nan's recipe
Why am I doing my Why = to seek Comfort

The why am I am I doing my what no longer makes sense.
The pattern is now.
_
.
.

Side view of the layers - What + how + why =
O o
O o .
O o .

If you've followed me on this thought experiment you might see how this helps me identify missing elements as I (theoretically i.e. with practise) am able to do a quick assessment of what I am doing, how, and why the pattern appears as it does (subjectively).

Furthering this - consider - what if I use a different framework? What if I add in another vector on layer 1? What might a matrix look like? How much recursion can these symbols handle? How does it differ from the way alchemy captured information? Would stereoscopic version add the elements of space and time?

This offers a way to visualise multilayer/dimensional mental blocks 🤩 I truly believe we need more ways for communicating complexity. Plus. It's fun.

So, to wrap up my project is, I want one ring to think it all!
Which fits, I do feel a bit like Gollum at times 😆

**Reply by jamesstephenbrown** [reply]
Oh wow, while I don't think I've entirely followed you, I do understand the impulse and the general idea, I can see how if a machine learning process were able create these vectors in a chunk like this, a data scientist could quite easily go in and interrogate the function of each chunk. I imagine you could have various aspects of each chunk linked to different aspects of other chunks so that the "why" of one chunk is closely correlated to the "how" of another that is otherwise unrelated.

This seems very much like the sorts of experiments I've done that are also very difficult to explain. I have for the last few years been obsessed with modelling information in a way that utilises multiple dimensions and finding ways to make that information readily understandable, including numerous self-organising node-graphs, to map language relationships (word predictors), multidimensional knowledge maps that try to both map relationships as well time and space (like historical movements, that have a time and a place but also a causal relationship with other events that are not aligned with time and space), and prioritisation maps to help with decision making (what task to do next based on numerous overlapping weights), as well as evolutionary algorithms for learning. And all my code is full of recursion - I find it sort of magical, the simplicity of it and the emergence of complexity it enables.

I think all of this effort actually lead to this representation of 3D tetris apparently popping into my head.

I do think it's interesting to think of all the ways we can perceive different qualities of information. 2 spacial dimensions through vision, a 3rd dimension through stereoscopy (because trying to perceive a 3rd dimension purely from a single perspective actually disrupts the 2d plane making the information messy), then there's colour, sound, touch? These can all be perceived simultaneously, and separately without overlapping (the obscuring problem), so we could potentially develop an understanding of a cluster that utilised these different separate perceptive channels that could represent something on ~6 dimensions, given enough training - not counting the dimension of time, which is a sort of dimension multiplier I would say.

Thanks for sharing, this is why I started the blog to have conversations like these. Hey have you checked out the latest blog, I'm starting a series on [alignment](https://nonzerosum.games/alignment1.html), the first one's pretty basic, but over the next few weeks I'll be fleshing out an idea I think is important but isn't really getting much attention, I'd love to hear your thoughts. My ideas aren't fully formed so feedback will actually help me create better subsequent posts :)

**Reply by FreeFallAnnie** [reply]
I laughed out loud at how ridiculously similar the trains we are on are.
Let me know if you want to chat about it over a call sometime!

I will check out alignment to offer my thoughts :D
